src/jit/lsra.cpp

```diff
@@ -4148,6 +4161,14 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
             useCandidates  = allRegs(registerType);
         }
 
+#ifdef ARM_SOFTFP
```

Comment:
At this point, the node has been created, so the `gtOtherRegs` values should be correct, right?  I think you could assert `tree->gtOtherRegs[i] == genRegNumFromMask(currCandidates)` below. It would also be good to add a comment that this code, and the code in lsraarmarch.cpp rely on the fact that the registers are in sequential order.

---
In reply to: [122643874](https://github.com/dotnet/coreclr/pull/12360#discussion_r122643874) [](ancestors = 122643874)